src/server/AppManager.ts

```diff
@@ -393,18 +393,32 @@ export class AppManager {
             settings: {},
             implemented: result.implemented.getValues(),
             marketplaceInfo,
-        });
+        };
+
+        // Now that is has all been compiled, let's get the
+        // the App instance from the source.
+        const app = this.getCompiler().toSandBox(this, compiled);
+
+        // Create an app user for this app before installing it.
+        const appUserId = await this.accessorManager.getModifier(app.getID()).getCreator().getUserCreator().createAppUser(app);
```

Comment:
The way this call is happening makes me think the App can make this call again. If that is true, is this a public method for the App and why is that so? I would think that this call would be an internal one. 🤔

Reply:
Yes, we should make this method private and also make sure it is called only once. I will change it ;) Thanks!

Reply:
Moved `createAppUser` to AppManager as a private method